tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Grpc/GrpcDotNet/GrpcAspNetCoreServer/GrpcDotNetServerCommon.cs

```diff
@@ -35,14 +35,12 @@ internal static class GrpcDotNetServerCommon
                 var tags = new GrpcServerTags();
                 GrpcCommon.AddGrpcTags(tags, tracer, method.GrpcType, name: method.Name, path: method.FullName, serviceName: method.ServiceName);
 
+                var extractedContext = ExtractPropagatedContext(requestMessage);
+                Baggage.Current.Merge(extractedContext.Baggage);
```

Comment:
Just to confirm, now we are doing the extraction unconditionally because we _always_ want to set baggage from the incoming headers, even if we are not using the trace context from the headers, correct?

Reply:
Correct. Baggage is independent of trace context. The incoming headers could contain only baggage and no trace context, for example, and we would still want to extract the baggage header.